lib/liquid/tags/render.rb

```diff
@@ -45,7 +45,7 @@ def initialize(tag_name, markup, options)
       @alias_name = Regexp.last_match(6)
       @variable_name_expr = variable_name ? parse_expression(variable_name) : nil
       @template_name_expr = parse_expression(template_name)
-      @for = (with_or_for == FOR)
+      @for_loop? = (with_or_for == FOR)
```

Comment:
I don't think instance variables can have a trailing question mark.  Instead of using `attr_reader`, you can just define a `for_loop?` method.  E.g.

```ruby
    def for_loop?
      @is_for_loop
    end
```

```suggestion
      @is_for_loop = (with_or_for == FOR)
```